src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -82,59 +82,78 @@ _hub switch
     /// <inheritdoc />
     public override void OnStart(Activity data)
     {
-        if (data.ParentSpanId != default && _map.TryGetValue(data.ParentSpanId, out var parentSpan))
+        if (data.ParentSpanId != default && _map.TryGetValue(data.ParentSpanId, out var mappedParent))
         {
-            // We can find the parent span - start a child span.
-            var context = new SpanContext(
-                data.OperationName,
-                data.SpanId.AsSentrySpanId(),
-                data.ParentSpanId.AsSentrySpanId(),
-                data.TraceId.AsSentryId(),
-                data.DisplayName,
-                null,
-                null)
-            {
-                Instrumenter = Instrumenter.OpenTelemetry
-            };
-
-            var span = (SpanTracer)parentSpan.StartChild(context);
-            span.StartTimestamp = data.StartTimeUtc;
-            // Used to filter out spans that are not recorded when finishing a transaction.
-            span.SetFused(data);
-            span.IsFiltered = () => span.GetFused<Activity>() is { IsAllDataRequested: false, Recorded: false };
-            _map[data.SpanId] = span;
+            // Explicit ParentSpanId of another activity that we have already mapped
+            CreateChildSpan(data, mappedParent, data.ParentSpanId);
+        }
+        else if (_hub.GetSpan() is IBaseTracer { IsOtelInstrumenter: false } inferredParent)
```

Comment:
I'm confused by this check.
`OnStart` handles now:
1. If the parent has already been mapped: Create a ChildSpan
2. The parent has not been mapped, we check the current span on the hub, and if it's not an Otel span (which means there has been a SentrySpan created somewhere, then we add a child to that.
3. Else we create a root span

Since this is strictly for mapping activities to spans, can we run into a scenario where the current span is an Otel span?

Reply:
I figured there were three scenarios.
1. Someone creates an OTel Span and then creates an OTel child span... all using OTel instrumentation. In this case, the ParentSpanId will be set on the child span. We'll find that parent span in our map (since it will have already passed through the span processor
2. Someone creates a Sentry transaction... then creates an OTel span. There won't be anything in the map that matches (as a parent) but we are operating in the context of a \[Sentry\] transaction... so we create a child of that transaction, even though we're instrumenting with OTel at this point
3. If neither of the above is the case, what we have is a new root span (aka transaction). 

The reason I check to make sure the current span is not OTel instrumented is that if there is an OTel instrumented span active and it's not explicitly marked as the parent of the Activity that our span processor is handling, then it's not supposed to be the parent span. They've intentionally instrumented things to start a new span tree at this point. I think [this](https://opentelemetry.io/docs/languages/net/instrumentation/#creating-new-root-activities) is the only way they could do that, but it is a logical possibility.

```csharp
public static void DoWork()
{
    var previous = Activity.Current;
    Activity.Current = null;

    var newRoot = MyActivitySource.StartActivity("NewRoot");

    // Re-set the previous Current Activity so the trace isn't messed up
    Activity.Current = previous;
}
```

Reply:
I'll add a comment in the code to hint at that.